dags/iop/parser.py

```diff
@@ -49,10 +54,24 @@ def _get_journal_doctype(self, article: ET.Element):
             self.logger.error("Article-type is not found in XML", doi=self.dois)
             return None
         try:
-            return self.article_type_mapping[value]
+            self.journal_doctype = self.article_type_mapping[value]
+            return self.journal_doctype
         except KeyError:
             self.logger.error(
                 "Unmapped article type", doi=self.dois, article_type=value
             )
         except Exception:
             self.logger.error("Unknown error", doi=self.dois)
+
+    def _get_related_article_doi(self, article):
+        if self.journal_doctype in ["corrigendum", "addendum"]:
+            node = article.find(
+                "front/article-meta/related-article[@ext-link-type='doi']"
+            )
+            if node is None:
+                return
+            related_article_doi = node.get("href")
+            if related_article_doi:
+                self.logger.info("Adding related_article_doi.")
+                return [related_article_doi]
```

Comment:
why you need to cast to list, can't it be just a `doi` value if it's just one?

Reply:
Good comment. Actually just because it was already implemented like this. Ricardo was keeping up with schema where dois is an array, but looks like he wasn't aware that related_article_doi + doi values will be put both in array. Fot his fix I can put dois as strings, however, we will need to change the code in generic parsing and dois/related_article_dois extractions in hindawi, aps and springer.

Reply:
@ErnestaP we can simplify the code:

```python
        if self.journal_doctype not in ["corrigendum", "addendum"]:
             return
        try:
            related_article_doi = article.find("front/article-meta/related-article[@ext-link-type='doi']").attrib['href']
            return [related_article_doi]
        except KeyError:
            self.logger.error("No related article dois found", dois=self.dois)
        
```

Reply:
f related_article_doi is None will we still have to return it?